pkg/services/app/add.go

```diff
@@ -692,14 +717,136 @@ func (a *App) createPullRequestToRepo(params AddParams, gitProvider gitproviders
 	}
 
 	userRepoRef := gitproviders.NewUserRepositoryRef(github.DefaultDomain, owner, repoName)
-	prLink, err := gitProvider.CreatePullRequestToUserRepo(userRepoRef, params.Branch, appHash, files, utils.GetCommitMessage(), fmt.Sprintf("wego add %s", params.Name), fmt.Sprintf("Added yamls for %s", params.Name))
+	prLink, err := gitProvider.CreatePullRequestToUserRepo(userRepoRef, info.Spec.Branch, appHash, files, utils.GetCommitMessage(), fmt.Sprintf("wego add %s", info.Name), fmt.Sprintf("Added yamls for %s", info.Name))
 	if err != nil {
 		return fmt.Errorf("unable to create pull request: %w", err)
 	}
 	a.logger.Println("Pull Request created: %s\n", prLink.Get().WebURL)
 	return nil
 }
 
+func (a *App) getAppResourceInfo(app wego.Application, clusterName string) *AppResourceInfo {
+	return &AppResourceInfo{
+		Application: app,
+		clusterName: clusterName,
+		targetName:  clusterName,
+	}
+}
+
+func (a *AppResourceInfo) automationRoot() string {
+	root := "."
+
+	if a.Spec.ConfigURL == string(ConfigTypeUserRepo) || a.Spec.ConfigURL == a.Spec.URL {
+		root = ".wego"
+	}
+
+	return root
+}
+
+func (a *AppResourceInfo) appYamlPath() string {
+	return filepath.Join(a.appYamlDir(), "app.yaml")
+}
+
+func (a *AppResourceInfo) appYamlDir() string {
+	return filepath.Join(a.automationRoot(), "apps", a.Name)
+}
+
+func (a *AppResourceInfo) appAutomationPath() string {
+	return filepath.Join(a.appAutomationDir(), fmt.Sprintf("%s-gitops-runtime.yaml", a.Name))
+}
+
+func (a *AppResourceInfo) appAutomationDir() string {
+	return filepath.Join(a.automationRoot(), "targets", a.clusterName, a.Name)
+}
+
+func (a *AppResourceInfo) sourceKind() string {
+	result := "GitRepository"
+
+	if a.Spec.SourceType == "helm" {
+		result = "HelmRepository"
+	}
+
+	return result
+}
+
+func (a *AppResourceInfo) sinkKind() string {
```

Comment:
What does `sink` mean in this case?

Reply:
Breaking each app automation into a "source" and a "sink" (input from repo, output to cluster). I'm happy to use some other terminology if you have any ideas.

Reply:
This appears to be returning an "automation" kind, which we are referring to as a `DeploymentType` elsewhere. Maybe `deployKind()` for now?

Reply:
Sure, that seems reasonable to me